src/foam/dao/PgConnectionPool.java

```diff
@@ -0,0 +1,32 @@
+package foam.dao;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public abstract class PgConnectionPool {
+
+    protected static String base = "jdbc:postgresql://";
```

Comment:
See FOAM style guide: https://github.com/foam-framework/foam/wiki/FOAM2-Style-Guidelines
Indentation is two spaces and spacing is like:
if ( foobar ) {